### PR TITLE
Adding isDynamic Support to SizeNodes

### PR DESCRIPTION
**Pull request by Gamrix**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77917
* #77916

### CODE REVIEW

File: torch/csrc/lazy/ts_backend/dynamic_ir.cpp [hunk 1]
```diff
@@ -36,6 +36,13 @@ SizeNode::SizeNode(Value input, size_t dim):
 int64_t SizeNode:: getStaticValue() const {
     return dynamic_cast<const TsNode*>(operand(0).node)->shape(0).size(dim_);
 }
+bool SizeNode:: isDynamic() const {
+    auto symbolic_vec = dynamic_cast<const TsNode*>(operand(0).node)->shape(0).is_symbolic();
+    if(!symbolic_vec.has_value()){
```
Comment (wconstab) [inline torch/csrc/lazy/ts_backend/dynamic_ir.cpp:41]: what's the reasoning behind this logic?

**Review by wconstab — COMMENTED** [review]
looks good to me, but let's make sure @Krovatkin also reviews

File: torch/csrc/lazy/ts_backend/dynamic_ir.cpp (hunk 1 above)
Comment (Gamrix) [reply torch/csrc/lazy/ts_backend/dynamic_ir.cpp:41]: The idea is that if the `symbolic_vec` is `nullopt`, it either is not calculatable, or it is an item with dynamic dims. The conservative thing to do in that case would be to mark it as symbolic.

Comment (Gamrix) [reply torch/csrc/lazy/ts_backend/dynamic_ir.cpp:41]: Not calculatable is either missing JIT functionality, and for the rare case where there are dynamic dims.

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Pull request closed without merging**